contract/src/assertions.rs

```diff
@@ -73,6 +73,43 @@ pub fn assert_mint_authority<'a>(
     Ok(())
 }
 
+// Asserts that the mint's account info is consistent with the data in
+// CreateTokenArgs on Token auction creation
+pub fn assert_token_mint_arg_consistency<'a>(
+    token_mint_info: &AccountInfo<'a>,
+    existing_account: &Option<Pubkey>,
+) -> Result<(), ProgramError> {
+    if (token_mint_info.data_is_empty() && !existing_account.is_none())
```

Comment:
maybe instead of `!x.is_none()` it's more clear to use `x.is_some()`. I would have expected clippy to pinpoint this.